[WebGPU] Remove release assert from WebGPU::createLibrary

mwyrzykowski opened the pull request:
#### 42d8933f0ae2fb8371afd2cc6dff3560e547d1a3
<pre>
[WebGPU] Remove release assert from WebGPU::createLibrary
<a href="https://bugs.webkit.org/show_bug.cgi?id=264025">https://bugs.webkit.org/show_bug.cgi?id=264025</a>
&lt;radar://117589524&gt;

Reviewed by Tadeu Zagallo.

Calling createLibrary with an invalid ShaderModule should
return std::nullopt otherwise the ast will not be valid.

* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::createLibrary):

Canonical link: <a href="https://commits.webkit.org/270965@main">https://commits.webkit.org/270965@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/32ec8db7c6bc5fcc6de2646f5d5a7b0932968237

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26941 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5557 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28177 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29152 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24631 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7412 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2949 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24504 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27203 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4380 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23127 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3834 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3903 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29787 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24585 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24530 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30119 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3920 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2117 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28030 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5379 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6467 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4386 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4289 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

tadeuzagallo commented on Source/WebGPU/WebGPU/Pipeline.mm:
```diff
@@ -42,7 +42,8 @@
     auto* ast = shaderModule.ast();
-    RELEASE_ASSERT(ast);
+    if (!ast)
+        return std::nullopt;
```
I guess I was going by a different approach, and making sure that `createLibrary` is never called with an invalid module, but I think either option is fine. FWIW we do return nullopt below, when we can't find the entry point.

mwyrzykowski replied:
Oh I'll do that instead. Indeed we shouldn't call this with an invalid shader module, maybe that is the root cause here

mwyrzykowski committed:
[WebGPU] Remove release assert from WebGPU::createLibrary
https://bugs.webkit.org/show_bug.cgi?id=264025
<radar://117589524>

Reviewed by Tadeu Zagallo.

Calling createLibrary with an invalid ShaderModule should
return std::nullopt otherwise the ast will not be valid.

* Source/WebGPU/WebGPU/Pipeline.mm:
(WebGPU::createLibrary):

Canonical link: https://commits.webkit.org/270965@main
